sysutils/libvirt/files/patch-st_birthtimespec.diff

```diff
@@ -0,0 +1,12 @@
+--- src/storage/storage_util.c	2024-06-18 07:39:04.000000000 +0800
++++ src/storage/storage_util.c	2024-06-18 07:33:34.000000000 +0800
+@@ -1898,7 +1898,9 @@
+ 
+ #ifdef __APPLE__
+     target->timestamps->atime = sb->st_atimespec;
++#if !defined(__ppc__) && !defined(__i386__)
+     target->timestamps->btime = sb->st_birthtimespec;
++#endif
```

Comment:
According to Apple documentation, [`st_birthtimespec`](https://developer.apple.com/documentation/kernel/stat64/1493635-st_birthtimespec) is available in Mac OS X 10.6 and later, so I doubt conditionalizing this on the architecture is correct. We had [a successful build on 10.6 i386](https://build.macports.org/builders/ports-10.6_i386-builder/builds/169974). Presumably you should conditionalize on the deployment target instead.

```suggestion
 #ifdef __APPLE__
 # include <AvailabilityMacros.h>
     target->timestamps->atime = sb->st_atimespec;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
     target->timestamps->btime = sb->st_birthtimespec;
+#endif
```

Reply:
@barracuda156 ?

Reply:
Yeah, I will return to this today.

Reply:
I'm not 100% certain whether it should be based on `MAC_OS_X_VERSION_MIN_REQUIRED` (the deployment target) or `MAC_OS_X_VERSION_MAX_ALLOWED` (the SDK version).

Reply:
@ryandesign It is a part of `stat64` though. 32-bit version does not have it, even in 10.6.8 SDK: https://developer.apple.com/documentation/kernel/stat (obviously, it failed on `ppc`, that is how I knew about the issue in the first place).